### ISSUE TITLE
How to run your own data set?

### ISSUE THREAD

**Issue by 1zgh** [facebookresearch/hyperreel#1]
Very good job!
How to run your own data set? colmap?
Thanks!!

**Comment by breuckelen** [comment]
Yep that's right! You can follow the instructions within the [LLFF readme](https://github.com/fyusion/llff#using-your-own-poses-without-running-colmap) to extract poses from your own dataset.

**Comment by 1zgh** [comment]
Thank you very much！
1、Yes, my scene is not forward facing. The scenes in my data set are shot in all directions. Which configuration files should I modify?
2、In addition, I have noticed that the llff dataset contains images that are subsampled and run out using colmap. But the original colmap does not produce files such as the poses_bounds.npy file. Do you have your own colmap script for everyone to use? thank you.

**Comment by ZhenyanSun** [comment]
> Thank you very much！ 1、Yes, my scene is not forward facing. The scenes in my data set are shot in all directions. Which configuration files should I modify? 2、In addition, I have noticed that the llff dataset contains images that are subsampled and run out using colmap. But the original colmap does not produce files such as the poses_bounds.npy file. Do you have your own colmap script for everyone to use? thank you.

Thanks for your question, and I also want to run on my own dataset, could you share more about what should I do?

**Comment by ZhenyanSun** [comment]
Thanks very much.

**Comment by 1zgh** [comment]
Thanks! I will try.
I see you mentioned distorted reflection and refraction, is this reconstructed for non-Lambertian objects? Mirrors, water surfaces, things like that. And there's an algorithmic optimization for that, right?

**Comment by breuckelen** [comment]
Yep! Given enough views, our method should be able to reproduce the appearance of such objects. For example see the reflections on the CD and refractions through the bottle of liquid in this scene:

https://user-images.githubusercontent.com/2993881/211887264-2874afc8-d4de-4990-a7cc-351c1903efcf.mov

**Comment by 1zgh** [comment]
1. I replicated the same effect in the CD dataset, which worked well.
But now we're having some trouble with our own datasets.
2. I have tried to use scenes that are inward facing (all cameras shooting around objects). The llff configuration file (run_one_llff.sh) is currently used in the training process, but the training loss and psnr effect are poor. When I tried to run my own data set using donerf's configuration file, I was prompted that many files were missing (sorry, I haven't used this data set before). Is there any detailed guidance for this type of scenario? Thanks!!
3. Here is the result of my llff configuration file, psnr=15.08. And you can only look forward when you're visualizing, whereas I shoot around objects. The data set used is the refnerf data set.
![gs1](https://user-images.githubusercontent.com/51021442/217991772-0f70198c-0c51-43c8-b6b6-211e8694d520.gif)

**Comment by breuckelen** [comment]
Is the from the "real" dataset provided on the Ref-NeRF website? If so, let me try our method on this data and get back to you.

Note that our method works quite well for *densely* captured scenes, but that it's difficult to capture inward facing 360 degree scenes with high angular density (so it might not work super well in this setting).

**Comment by 1zgh** [comment]
Yes! It's a real data set from the website.  I'm going to try something similar.
That it's difficult to capture inward facing 360 degree scenes with high angular density (so it might not work super well in this setting).Could you please tell me the reason for this result? geometric primitives?

**Comment by breuckelen** [comment]
Yep, because of the sample prediction network, which predicts geometric primitives / sample points that vary depending on the input (4D) ray. This allows the framework to reproduce complex view dependence, but also means that the samples are not *explicitly constrained* for rays that are not observed during training (and so it tends to lead to worse interpolation / extrapolation quality in sparse view settings).

This is one of the main limitations of the work, which we discuss in the conclusion of the paper:

![image](https://user-images.githubusercontent.com/2993881/218018701-06618103-4d52-4918-a83f-bb78238b162b.png)

**Comment by 1zgh** [comment]
thank! I see. But the experimental scenes in real nerf are still curious.

**Comment by breuckelen** [comment]
No problem! And I agree -- it's still definitely worth looking into.

**Comment by 1zgh** [comment]
I am very willing to work with you on this work. Is there anything I can do to help speed up the progress of this work?
